yamlpath/enums/pathseperators.py

```diff
@@ -1,120 +1,10 @@
-
-    `DOT`
-        YAML Path segments are seperated via dots (.).
-
-    `FSLASH`
-        YAML Path segments are seperated via forward-slashes (/).
-    """
-
-    AUTO = auto()
-    DOT = auto()
-    FSLASH = auto()
-
-    def __str__(self) -> str:
-        """Get a String representation of this employed enum's value."""
-        seperator = '.'
-        if self is PathSeperators.FSLASH:
-            seperator = '/'
-        return seperator
-
-    @staticmethod
-    def get_names() -> List[str]:
-        """
-        Get all upper-cased entry names for this enumeration.
-
-        Parameters:  N/A
-
-        Returns:  (List[str]) Upper-case names from this enumeration
-
-        Raises:  N/A
-        """
-        return [entry.name.upper() for entry in PathSeperators]
-
-    @staticmethod
-    def get_choices() -> List[str]:
-        """
-        Get all entry names with symbolic representations for this enumeration.
-
-        All returned entries are lower-cased.
-
-        Parameters:  N/A
-
-        Returns:  (List[str]) Lower-case names and symbols from this
-            enumeration
-
-        Raises:  N/A
-        """
-        names = [l.lower() for l in PathSeperators.get_names()]
-        symbols = [str(e) for e in PathSeperators]
-        choices = list(set(names + symbols))
-        choices.sort()
-        return choices
-
-    @staticmethod
-    def from_str(name: str) -> "PathSeperators":
-        """
-        Convert a string value to a value of this enumeration, if valid.
-
-        Parameters:
-            1. name (str) The name to convert
-
-        Returns:  (PathSeperators) the converted enumeration value
-
-        Raises:
-            - `NameError` when name doesn't match any enumeration values
-        """
-        if isinstance(name, PathSeperators):
-            return name
-
-        check: str = str(name).upper()
-        if check == '.':
-            check = "DOT"
-        elif check == '/':
-            check = "FSLASH"
-
-        if check in PathSeperators.get_names():
-            return PathSeperators[check]
-        raise NameError("PathSeperators has no such item, {}.".format(check))
-
-    @staticmethod
-    def infer_seperator(yaml_path: str) -> "PathSeperators":
-        """
-        Infer the seperator used within a sample YAML Path.
-
-        Will attempt to return the best PathSeperators match.  Always returns
-        `PathSeperators.AUTO` when the sample is empty.
-
-        Parameters:
-            1. yaml_path (str) The sample YAML Path to evaluate
-
-        Returns: (PathSeperators) the inferred PathSeperators value
-
-        Raises:  N/A
-        """
-        seperator: PathSeperators = PathSeperators.AUTO
-
-        if yaml_path:
-            if yaml_path[0] == '/':
-                seperator = PathSeperators.FSLASH
-            else:
-                seperator = PathSeperators.DOT
-
-        return seperator
+from .pathseparators import PathSeparators as PathSeperators
```

Comment:
OK, thank you, let's figure out the breakage paths.

The above sample alone works fine:

```python
In [1]: from yamlpath.enums import PathSeperators

In [2]: sep = PathSeperators.DOT

In [3]: sep
Out[3]: <PathSeparators.DOT: 2>
```

The only problems I immediately see are for code that explicitly does type inspection, for example with `isinstance`, and then any development-level checking (e.g. mypy). To preserve that behavior, could/should we rearrange things so that the `PathSeparators` enum inherits from `PathSeperators`?

Reply:
Never in my wildest dreams did I imagine Python would auto-translate as it did between lines 2 and 3 in your output!  This revelation got me searching around Google trying to understand this behavior and I'm coming up with mixed support and open code review requests against Python itself to support Enum subclassing versus inheritance, etc.  This seems like an unsettled behavior unless I'm just not finding hard support that you know of.  Do you have a documentation reference (that my Google-fu is failing to find) that specifies this is the expected behavior?

If so, we'll codify your solution one way or the other (PathSeperators inherits from PathSeparators or the other way around) by adding some novel tests to ensure both enums are supported in future releases.  If not, we'll be forced to assume this is an undocumented feature of Python that can stop working at any time; in which case, we'll need to explicitly support both spellings and use Unions to accept either as input to affected methods, deprecating the bad spelling this release, targeting formal removal in 4.x.

Reply:
It's entirely possible that I'm just looking at this entirely too closely and not seeing the obvious.  :)

Reply:
Just adding for now that we may have no need for inheritance/subclassing anyway:

```python
In [1]: from yamlpath.enums import PathSeperators

In [2]: sep = PathSeperators.DOT

In [3]: sep
Out[3]: <PathSeparators.DOT: 2>

In [4]: type(sep)
Out[4]: <enum 'PathSeparators'>

In [5]: isinstance(sep, PathSeperators)
Out[5]: True

In [6]: from yamlpath.enums import PathSeparators

In [7]: isinstance(sep, PathSeparators)
Out[7]: True

In [8]: sepa = PathSeparators.FSLASH

In [9]: isinstance(sepa, PathSeperators)
Out[9]: True

In [10]: isinstance(sepa, PathSeparators)
Out[10]: True

```

Reply:
I just added in compatibility and coverage for explicit use of `infer_seperator`:

```diff
diff --git a/tests/test_enums_pathseparators.py b/tests/test_enums_pathseparators.py
index e65969b..fc2e795 100644
--- a/tests/test_enums_pathseparators.py
+++ b/tests/test_enums_pathseparators.py
@@ -48,5 +48,6 @@ class Test_enums_PathSeparators():
 		("/abc/123", PathSeparators.FSLASH),
 	])
 	@pytest.mark.parametrize("pathsep_module", [PathSeparators, PathSeperators])
-	def test_infer_separator(self, input, output, pathsep_module):
-		assert output == pathsep_module.infer_separator(input)
+	@pytest.mark.parametrize("func_name", ["infer_separator", "infer_seperator"])
+	def test_infer_separator(self, input, output, pathsep_module, func_name):
+		assert output == getattr(pathsep_module, func_name)(input)
diff --git a/yamlpath/enums/pathseparators.py b/yamlpath/enums/pathseparators.py
index e5980db..3b34835 100644
--- a/yamlpath/enums/pathseparators.py
+++ b/yamlpath/enums/pathseparators.py
@@ -118,3 +118,24 @@ class PathSeparators(Enum):
                 separator = PathSeparators.DOT
 
         return separator
+
+    @staticmethod
+    def infer_seperator(yaml_path: str) -> "PathSeparators":
+        """
+        Infer the separator used within a sample YAML Path.
+
+        Will attempt to return the best PathSeparators match.  Always returns
+        `PathSeparators.AUTO` when the sample is empty.
+
+        This is provided for compatibility with older versions,
+        before the spelling was updated to "separator."
+
+        Parameters:
+            1. yaml_path (str) The sample YAML Path to evaluate
+
+        Returns: (PathSeparators) the inferred PathSeparators value
+
+        Raises:  N/A
+        """
+
+        return PathSeparators.infer_separator(yaml_path)

```

With this, I think backward compatibility is in a very good state. Locally my eyaml-related tests don't work (even in development branch), but AFAICT the code in this branch passes all tests as they are in the development branch.

Reply:
I do think you're overthinking the enum compatibility; there's only one enum/type, available via a couple of names.

Reply:
Python isn't my only programming language.  I flip-flop between it and other languages with vastly more strict typing.  That just had my mind in a state that freaked out over such loose definition of the alternate name.